image.go

```diff
@@ -123,50 +126,79 @@ func (t MediaTypes) LayerType() types.MediaType {
 
 // OverrideMediaTypes mutates the provided v1.Image to use the desired media types
 // in the image manifest and config files (including the layers referenced in the manifest)
-func OverrideMediaTypes(base v1.Image, mediaTypes MediaTypes) (v1.Image, error) {
+func OverrideMediaTypes(image v1.Image, mediaTypes MediaTypes) (v1.Image, error) {
 	if mediaTypes == DefaultTypes || mediaTypes == MissingTypes {
 		// without media types option, default to original media types
-		return base, nil
+		return image, nil
 	}
 
 	// manifest media type
-	image := mutate.MediaType(empty.Image, mediaTypes.ManifestType())
-	// update empty image with base config
-	config, err := base.ConfigFile()
+	retImage := mutate.MediaType(empty.Image, mediaTypes.ManifestType())
+
+	// update empty image with image config
+	config, err := image.ConfigFile()
 	if err != nil {
 		return nil, err
 	}
+	history := config.History
+	// zero out diff IDs and history, as these will be updated when we call `mutate.Append`
 	config.RootFS.DiffIDs = make([]v1.Hash, 0)
-	image, err = mutate.ConfigFile(image, config)
+	config.History = []v1.History{}
+	retImage, err = mutate.ConfigFile(retImage, config)
 	if err != nil {
 		return nil, err
 	}
 
 	// config media type
-	image = mutate.ConfigMediaType(image, mediaTypes.ConfigType())
+	retImage = mutate.ConfigMediaType(retImage, mediaTypes.ConfigType())
 
 	// layers media type
-	layers, err := base.Layers()
+	layers, err := image.Layers()
 	if err != nil {
 		return nil, err
 	}
-	additions := layersAddendum(layers, mediaTypes.LayerType())
-	image, err = mutate.Append(image, additions...)
+	additions := layersAddendum(layers, history, mediaTypes.LayerType())
+	retImage, err = mutate.Append(retImage, additions...)
 	if err != nil {
 		return nil, err
 	}
 
-	return image, nil
+	return retImage, nil
+}
+
+// OverrideHistoryIfNeeded zeroes out the history if the number of history entries doesn't match the number of layers.
+func OverrideHistoryIfNeeded(image v1.Image) (v1.Image, error) {
+	configFile, err := image.ConfigFile()
+	if err != nil {
+		return nil, err
+	}
+	layers, err := image.Layers()
+	if err != nil {
+		return nil, err
+	}
+	retImage := image
```

Comment:
ok this is maybe my own confusion(s) here.  but `v1.Image` is (probably) an interface, right?  which means here you're doing a pointer copy, right?  so below when you call `mutate` on the `retImage` doesn't that _also_ change the `image` param that was passed in?   (that would be a reason to do a deepcopy here,  imo, if you're trying not to change your input param)

Reply:
Summarizing offline discussion - I think we agreed that this is weird, but it is the way to do it unless you want to use a pointer to interface, which is also weird